src/AutoMapper/Bundle/DependencyInjection/JaneAutoMapperExtension.php

```diff
@@ -61,22 +61,11 @@ public function load(array $configs, ContainerBuilder $container)
 
     private function createMapperConfigurationDefinition(ContainerBuilder $container, $config)
     {
-        $serviceName = 'Mapping_' . $config['source'] . '_' . $config['target'];
-        $definition = $container->register($serviceName, MapperMetadata::class);
-        $definition->setFactory([new Reference(MapperGeneratorMetadataFactory::class), 'create']);
-        $definition->addArgument(new Reference(AutoMapper::class));
+        $serviceName = 'LegacyDecorator_' . $config['pass'];
+        $definition = $container->register($serviceName, LegacyConfigurationPassDecorator::class);
+        $definition->addArgument(new Reference($config['pass']));
         $definition->addArgument($config['source']);
         $definition->addArgument($config['target']);
-        $definition->addTag('jane_auto_mapper.mapper_metadata');
-
-        if (\array_key_exists('pass', $config)) {
-            $restrictConfigurationPass = $container->register($serviceName . '_RestrictPass', RestrictConfigurationPass::class);
-            $restrictConfigurationPass->addArgument($config['source']);
-            $restrictConfigurationPass->addArgument($config['target']);
-            $restrictConfigurationPass->addArgument(new Reference($config['pass']));
-
-            $definition = $container->getDefinition(AutoMapper::class);
-            $definition->addMethodCall('addConfigurationPass', [new Reference($serviceName . '_RestrictPass')]);
-        }
```

Comment:
I would like to keep the `RestrictConfigurationPass` behavior (which check if given metadata match given source/target), I think it's still needed.

Reply:
With the current implementation, it's not needed as the correct metadata is built when we add a new `MapperConfigurationInterface` (see [here](https://github.com/janephp/janephp/pull/453/files#diff-99fbbcf461f1d70369c81053b1f9d5f5ba9dd5b1dcec6c14123481fab69e5637R12-R13))